crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -38,6 +41,263 @@ async fn local_node_price_improvement_fee_sell_order() {
+        policy: FeePolicyKind::PriceImprovement {
+            factor: 0.3,
+            max_volume_factor: 0.9,
+        },
+        policy_order_class: FeePolicyOrderClass::Market,
+    };
+    let partner_fee_app_data = OrderCreationAppData::Full {
+        full: json!({
+            "version": "1.1.0",
+            "metadata": {
+                "partnerFee": {
+                    "bps":1000,
+                    "recipient": "0xb6BAd41ae76A11D10f7b0E664C5007b908bC77C9",
+                }
+            }
+        })
+        .to_string(),
+    };
+
+    let autopilot_config = vec![
+        ProtocolFeesConfig(vec![limit_surplus_policy, market_price_improvement_policy]).to_string(),
+        "--fee-policy-max-partner-fee=0.02".to_string(),
+    ];
+
+    let mut onchain = OnchainComponents::deploy(web3.clone()).await;
+
+    let [solver] = onchain.make_solvers(to_wei(200)).await;
+    let [trader] = onchain.make_accounts(to_wei(200)).await;
+    let [limit_order_token, market_order_token, partner_fee_order_token] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(10), to_wei(10))
+        .await;
+
+    limit_order_token.mint(solver.address(), to_wei(1000)).await;
+    market_order_token.mint(solver.address(), to_wei(100)).await;
+    partner_fee_order_token
+        .mint(solver.address(), to_wei(100))
+        .await;
+    tx!(
+        solver.account(),
+        limit_order_token.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        market_order_token.approve(onchain.contracts().uniswap_v2_router.address(), to_wei(100))
+    );
+    tx!(
+        solver.account(),
+        partner_fee_order_token
+            .approve(onchain.contracts().uniswap_v2_router.address(), to_wei(100))
+    );
+    tx!(
+        trader.account(),
+        onchain
+            .contracts()
+            .weth
+            .approve(onchain.contracts().allowance, to_wei(100))
+    );
+    tx_value!(
+        trader.account(),
+        to_wei(100),
+        onchain.contracts().weth.deposit()
+    );
+    tx!(
+        solver.account(),
+        onchain
+            .contracts()
+            .weth
+            .approve(onchain.contracts().uniswap_v2_router.address(), to_wei(200))
+    );
+
+    let services = Services::new(onchain.contracts()).await;
+    let solver_endpoint =
+        colocation::start_baseline_solver(onchain.contracts().weth.address()).await;
+    colocation::start_driver(
+        onchain.contracts(),
+        vec![SolverEngine {
+            name: "test_solver".into(),
+            account: solver.clone(),
+            endpoint: solver_endpoint,
+        }],
+    );
+    services
+        .start_api(vec![
+            "--price-estimation-drivers=test_quoter|http://localhost:11088/test_solver".to_string(),
+        ])
+        .await;
+
+    let market_price_improvement_order = OrderCreation {
```

Comment:
How do I know this is a market order? I think it would be nice to quote this order and then use limits that are worse than the quote.

Reply:
Used a slightly decreased value from the quote.